tests/tests_sign_workflow/test_sign_artifacts.py

```diff
@@ -0,0 +1,40 @@
+import unittest
+from pathlib import Path
+
+from sign_workflow.sign_artifacts import SignArtifacts, SignArtifactsExistingArtifactFile, SignExistingArtifactsDir, SignWithManifest
+
+
+class TestSignArtifacts(unittest.TestCase):
+
+    def test_signer_class_method(self):
```

Comment:
Remove `_class_method` from the name of the function, use the actual function name.

Reply:
The name of the method is `__signer_class__`

Reply:
It would then be called `test_signer_class`.

You should use `assertIs` to avoid to do `type` twice.

Using variables that aren't actually reused is an anti-pattern.

Maybe this reads cleaner?

```python
    def test_signer_class(self):
        self.assertIs(SignArtifacts.__signer_class__(
            Path(r"/dummy/path/manifest.yml")), 
            SignWithBuildManifest)
            
        self.assertIs(SignArtifacts.__signer_class__(
            Path(os.path.dirname(__file__))), 
            SignExistingArtifactsDir)

        self.assertIs(SignArtifacts.__signer_class__(
            Path(r"/dummy/path/artifact.tar.gz")), 
            SignArtifactsExistingArtifactFile)
```